docs/frontend-performance.md

```diff
@@ -0,0 +1,2984 @@
+
+  const totalBalance = useMemo(
+    () => accounts.reduce((sum, acc) => sum + balances[acc.address], 0),
+    [accounts, balances]
+  );
+};
+```
+
+**Why:** Reselect selectors defined in other files are opaque to React Compiler. However, Reselect already provides memoization, so this is often not an issue.
+
+#### 8. Effect Dependencies (Keep Existing Memoization)
+
+**Effects memoization is still an open area of research.** React Compiler may memoize differently than manual memoization, which can break effects that rely on stable dependencies.
+
+```typescript
+⚠️ IMPORTANT: Keep existing useMemo/useCallback for effect dependencies
+const TokenBalance = ({ address }: Props) => {
+  const [balance, setBalance] = useState('0');
+  const network = useSelector(getNetwork);
+
+  // Keep useMemo to ensure effect behavior is preserved
+  const networkConfig = useMemo(() => ({
+    chainId: network.chainId,
+    rpcUrl: network.rpcUrl,
+  }), [network.chainId, network.rpcUrl]);
+
+  useEffect(() => {
+    // Stable networkConfig reference prevents infinite loops
+    fetchBalance(address, networkConfig);
+  }, [address, networkConfig]);
+};
+```
+
+**Why:** React Compiler will statically validate that auto-memoization matches existing manual memoization. If it can't prove they're the same, it safely skips compilation. To ensure effect behavior doesn't change, **keep existing `useMemo`/`useCallback` calls for effect dependencies**.
+
+**Recommendation:**
+
+- ✅ Keep existing `useMemo`/`useCallback` for effects
+- ✅ Write new code without manual memoization (let React Compiler handle it)
+- ⚠️ If you notice unexpected effect behavior, [file an issue](https://github.com/facebook/react/issues)
+
+#### 9. Context Values from External Providers
+
+When consuming context from providers defined in other files, React Compiler may not optimize effectively.
+
+```typescript
+❌ WRONG: React Compiler may not optimize context from other files
+// context.tsx (different file)
+export const ExternalI18nContext = createContext(/* ... */);
+
+// component.tsx
+import { ExternalI18nContext } from './context';
+
+const TokenDisplay = ({ token }: Props) => {
+  const { formatCurrency, locale } = useContext(ExternalI18nContext);
+
+  // React Compiler may not optimize context values from another file
+  const formattedBalance = formatCurrency(token.balance, locale);
+};
+
+✅ CORRECT: Manual memoization if needed
+const TokenDisplay = ({ token }: Props) => {
+  const { formatCurrency, locale } = useContext(ExternalI18nContext);
+
+  // Manual memoization if this computation is expensive
+  const formattedBalance = useMemo(
+    () => formatCurrency(token.balance, locale),
+    [formatCurrency, token.balance, locale]
+  );
+};
+```
+
+**Why:** Context providers defined in other files may not be fully analyzed by React Compiler. However, simple context consumption often works fine without manual memoization.
+
+#### 10. Computations with Multiple Cross-File Dependencies
```

Comment:
This doesn't seem to be too much different from `1. Cross-File Dependencies` and few other points already mentioned. It might be redundant.